lib/kcerts/ked25519/cert.go

```diff
@@ -0,0 +1,87 @@
+package ked25519ackage
+
+import (
+	"golang.org/x/crypto/ed25519"
+	"golang.org/x/crypto/ssh"
+	"math/rand"
+)
+// Copied from https://github.com/mikesmitty/edkey/blob/master/edkey.go
+/* Writes ed25519 private keys into the new OpenSSH private key format.
+I have no idea why this isn't implemented anywhere yet, you can do seemingly
+everything except write it to disk in the OpenSSH private key format. */
+func MarshalED25519PrivateKey(key ed25519.PrivateKey) []byte {
+	// Add our key header (followed by a null byte)
+	magic := append([]byte("openssh-key-v1"), 0)
+
+	var w struct {
+		CipherName   string
+		KdfName      string
+		KdfOpts      string
+		NumKeys      uint32
+		PubKey       []byte
+		PrivKeyBlock []byte
+	}
+
+	// Fill out the private key fields
+	pk1 := struct {
+		Check1  uint32
+		Check2  uint32
+		Keytype string
+		Pub     []byte
+		Priv    []byte
+		Comment string
+		Pad     []byte `ssh:"rest"`
+	}{}
+
+	// Set our check ints
+	ci := rand.Uint32()
+	pk1.Check1 = ci
+	pk1.Check2 = ci
+
+	// Set our key type
+	pk1.Keytype = ssh.KeyAlgoED25519
+
+	// Add the pubkey to the optionally-encrypted block
+	pk, ok := key.Public().(ed25519.PublicKey)
+	if !ok {
+		//fmt.Fprintln(os.Stderr, "ed25519.PublicKey type assertion failed on an ed25519 public key. This should never ever happen.")
```

Comment:
this function should probably return `([]byte, error)`, this should be a `fmt.Errorf()`

Reply:
well, ok, nil is ok, as a single error is possible. We should add to the comment documenting the function that it will either return a byte array, or nil in case of error.

Reply:
the // fmt.Println can go away.